src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -158,14 +158,25 @@ public override object Invoke(BindingFlags invokeAttr, Binder? binder, object?[]
                 throw new TargetParameterCountException(SR.Arg_ParmCnt);
             }
 
+            // Array construction is a special case.
+            if (DeclaringType is not null && DeclaringType.IsArray)
+            {
+                Span<int> args = stackalloc int[actualCount];
+                for (int i = 0; i < actualCount; ++i)
+                {
+                    args[i] = (int)parameters![i]!;
```

Comment:
The existing implementation does implicit widening, like all other reflection invoke APIs. For example, it is possible to reflection invoke the method with `byte` arguments that will be widened to `int`. This does not do the implicit widening anymore.

Reply:
Good call. Let me see what I can do here.

Reply:
This proved to not be a good use of my time. I am simply moving some of this into a separate code path to declutter the most common `Invoke` code path.